### PR TITLE
feat(nx): add ability to ignore files from affected commands

### PR DESCRIPTION
**Pull request by wesleygrimes**
Check .nxignore for globs to add to ignored files in affected commands

_[Please make sure you have read the submission guidelines before posting an PR](https://github.com/nrwl/nx/blob/master/CONTRIBUTING.md#submit-pr)_

> _Please make sure that your commit message follows our format._

> Example: `fix(nx): must begin with lowercase`

## Current Behavior (This is the behavior we have today, before the PR is merged)

Affected commands ignore files/folders in .gitignore

## Expected Behavior (This is the new behavior we can expect after the PR is merged)

Affected commands ignore files/folders in .gitignore and .nxignore

## Issue

Implement #895

### CODE REVIEW

**Comment by wesleygrimes** [comment]
This is WIP, need to update documentation to mention new ability to provide `.nxignore` and `nx.ignore` property in `package.json`.

File: docs/angular/fundamentals/develop-like-google.md [hunk 1]
```diff
@@ -341,6 +342,18 @@ The `tags` array is used to impose constraints on the dependency graph. Read mor
 
 Nx uses its advanced code analysis to construct a dependency graph of all applications and libraries. Some dependencies, however, cannot be determined statically. You can use the `implicitDependencies` array to list the dependencies that cannot be determined statically.
 
+The `ignore` property is _optional_, but can defined to ignore additional files and folders during the `affected:*` commands.
```
Comment (vsavkin) [inline docs/angular/fundamentals/develop-like-google.md]: you should update the react and web guides as well

Comment (wesleygrimes) [reply docs/angular/fundamentals/develop-like-google.md]: Will do

**Comment by wesleygrimes** [comment]
@vsavkin updated `react` and `web` guides to mention ignore feature

**Comment by wesleygrimes** [comment]
Looks like the build/test on travis failing because of some cached files for cypress.

**Comment by brandonroberts** [comment]
You can delete the cache for your PR by clicking on the "More Options" menu and going to caches

**Comment by wesleygrimes** [comment]
> You can delete the cache for your PR by clicking on the "More Options" menu and going to caches. Then restart the job

Looks like I don't have access rights to do that. Unless I am missing something.

Screen Shot 2019-07-17 at 9 05 44 AM

**Comment by vsavkin** [comment]
@wesleygrimes could you rebase it? Sorry about it.

**Comment by wesleygrimes** [comment]
> @wesleygrimes could you rebase it? Sorry about it.

Sure. Will do.

File: docs/angular/fundamentals/monorepos-automation.md [hunk 2]
```diff
@@ -341,6 +342,18 @@ The `tags` array is used to impose constraints on the dependency graph. Read mor
 
 Nx uses its advanced code analysis to construct a dependency graph of all applications and libraries. Some dependencies, however, cannot be determined statically. You can use the `implicitDependencies` array to list the dependencies that cannot be determined statically.
 
+The `ignore` property is _optional_, but can defined to ignore additional files and folders during the `affected:*` commands.
```
Comment (vsavkin) [inline docs/angular/fundamentals/monorepos-automation.md]: I'd remove this from this doc. It's a fundamentals doc that is supposed to give an overview of the whole monorepo story. I think the `ignore` is a bit more advanced. And just seeing the ignore property gives folks enough of an idea that the feature is there.

File: docs/web/fundamentals/monorepos-automation.md [hunk 3]
```diff
@@ -342,6 +343,18 @@ The `tags` array is used to impose constraints on the dependency graph. Read mor
 
 Nx uses its advanced code analysis to construct a dependency graph of all applications and libraries. Some dependencies, however, cannot be determined statically. You can use the `implicitDependencies` array to list the dependencies that cannot be determined statically.
 
+The `ignore` property is optional, but can defined to ignore additional files and folders during the `affected:*` commands.
```
Comment (vsavkin) [inline docs/web/fundamentals/monorepos-automation.md]: Same as above

**Review by vsavkin — CHANGES_REQUESTED** [review]
I don't think we should support both .nxignore and the ignore property in nx.json. They do the same thing.

I'd just add support for .nxignore. Then, the PR will be a lot smaller.

**Comment by wesleygrimes** [comment]
Ready for review

**Comment by johannesschobel** [comment]
This, somehow does not work for me. I added a `.nxignore` file in the root-folder of my workspace and added the line `**/generated` in order to **exclude** all files in `generated` folders in my projects.
While this works for the `.gitignore`, the `generated/*.ts` files are still linted during my `npm run affected:lint` command - resulting in errors :(

Any solution for this?

**Pull request merged**